pkg/controllermanager/controller/namespacedcloudprofile/reconciler_test.go

```
@@ -809,5 +809,59 @@ var _ = Describe("NamespacedCloudProfile Reconciler", func() {
 			})
 		})
+
+		Describe("#MergeCloudProfiles", func() {
+			Describe("merge limits.maxNodesTotal correctly", func() {
+				var (
+					cloudProfile           *gardencorev1beta1.CloudProfile
+					namespacedCloudProfile *gardencorev1beta1.NamespacedCloudProfile
+				)
+
+				BeforeEach(func() {
+					cloudProfile = &gardencorev1beta1.CloudProfile{}
+					namespacedCloudProfile = &gardencorev1beta1.NamespacedCloudProfile{}
+				})
+
+				It("should stay nil if no value is provided at all", func() {
+					namespacedcloudprofilecontroller.MergeCloudProfiles(namespacedCloudProfile, cloudProfile)
+
+					Expect(namespacedCloudProfile.Status.CloudProfileSpec.Limits).To(BeNil())
+				})
+
+				It("should apply the only value from the CloudProfile", func() {
+					cloudProfile.Spec.Limits = &gardencorev1beta1.Limits{MaxNodesTotal: ptr.To(int32(10))}
+
+					namespacedcloudprofilecontroller.MergeCloudProfiles(namespacedCloudProfile, cloudProfile)
+
+					Expect(namespacedCloudProfile.Status.CloudProfileSpec.Limits.MaxNodesTotal).To(Equal(ptr.To(int32(10))))
+				})
+
+				It("should apply the only value from the NamespacedCloudProfile", func() {
+					namespacedCloudProfile.Spec.Limits = &gardencorev1beta1.Limits{MaxNodesTotal: ptr.To(int32(10))}
+
+					namespacedcloudprofilecontroller.MergeCloudProfiles(namespacedCloudProfile, cloudProfile)
+
+					Expect(namespacedCloudProfile.Status.CloudProfileSpec.Limits.MaxNodesTotal).To(Equal(ptr.To(int32(10))))
+				})
+
+				It("should ignore a higher value from the NamespacedCloudProfile", func() {
+					cloudProfile.Spec.Limits = &gardencorev1beta1.Limits{MaxNodesTotal: ptr.To(int32(10))}
+					namespacedCloudProfile.Spec.Limits = &gardencorev1beta1.Limits{MaxNodesTotal: ptr.To(int32(99))}
+
+					namespacedcloudprofilecontroller.MergeCloudProfiles(namespacedCloudProfile, cloudProfile)
+
+					Expect(namespacedCloudProfile.Status.CloudProfileSpec.Limits.MaxNodesTotal).To(Equal(ptr.To(int32(10))))
+				})
```

Is a higher value in the NSCPFL silently ignored?

---

Yes, that is the case.

---

That means the UX in the case of an unintentional misconfiguration is a bit confusing. You might be wondering why your namespaced cloud profile does not have any effect. Could this be reflected in the status of the resource?

---

The limits-override in a nscpfl should only be relevant in expert-usecases; also the field docs state:
> // If a limit is already set in the parent CloudProfile, it can only be more restrictive in the NamespacedCloudProfile.
Hence, I currently do not see the need for any message in the status or an api-client warning.
WDYT @timuthy? 😃

---

The implementation should be easy in the NSCPFL admission plugin, right? In this case I'd prefer the validation.

---

Done with [2dcca28](https://github.com/gardener/gardener/pull/11647/commits/2dcca286f9c21191847503cc3f9d65cf4d1c54b2).